File: src/main/java/com/matag/game/turn/TurnRequest.java
package com.matag.game.turn;

import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
public class TurnRequest {
  String action;
  String inputRequiredAction;
  String inputRequiredActionParameter;
  String inputRequiredChoices;
  Map<Integer, List<String>> mana;
  List<Integer> cardIds;
  Map<Integer, List<Object>> targetsIdsForCardIds;
  String playedAbility;
}

File: src/main/kotlin/com/matag/game/turn/TurnRequest.kt
package com.matag.game.turn

data class TurnRequest(
    var action: String? = null,
    var inputRequiredAction: String? = null,
    var inputRequiredActionParameter: String? = null,
    var inputRequiredChoices: String? = null,
    var mana: Map<Int, List<String>> = mapOf(),
    var cardIds: List<Int> = listOf(),
    var targetsIdsForCardIds: Map<Int, List<Any>> = mapOf(),
    var playedAbility: String? = null,
)
